### ISSUE TITLE
Crash when calling rs.getInt()

### ISSUE THREAD

**Issue by jacob-lerenius** [xerial/sqlite-jdbc#212]
[hs_err_pid168396.txt](https://github.com/xerial/sqlite-jdbc/files/896225/hs_err_pid168396.txt)
We get random crashes when performing a simple getInt() on the result set. Attaching the crash log.
Have tried various ways to handle this, but no luck, no exceptions so hard to understand what is going wrong.

**Comment by RonZhong2009** [comment]
I can't open the your attached file. But from your descriptions, seems we have encountered the same issue. I upgraded to the latest 3.19.3, the issue still got hit. Here is my stackframes about it:

> Stack: [0x000000006bdf0000,0x000000006bef0000],  sp=0x000000006bee8160,  free space=992k
> Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native code)
> C  [sqlite-3.19.3-1b1aca0b-79af-4746-ac09-72cc0a3c9878-sqlitejdbc.dll+0x122d4]
> C  [sqlite-3.19.3-1b1aca0b-79af-4746-ac09-72cc0a3c9878-sqlitejdbc.dll+0x1f76e]
> C  0x00000000045635fb
> 
> Java frames: (J=compiled Java code, j=interpreted, Vv=VM code)
> J 8999  org.sqlite.core.NativeDB.column_int(JI)I (0 bytes) @ 0x00000000045634bb [0x0000000004563480+0x3b]
> J 9581 C1 org.sqlite.jdbc3.JDBC3ResultSet.getInt(Ljava/lang/String;)I (10 bytes) @ 0x0000000003a2d4d4 [0x0000000003a2cfa0+0x534]

Any updates regarding this issue? Thanks in advance.

**Comment by jacob-lerenius** [comment]
I found out this happened when another thread closed the connection to the
DB before accessing rs.getInt(). Even though it was an error to do so, I
don't think this should cause a crash, better to throw an exception.
Best Regards
Jacob

Den 28 juni 2017 11:44 fm skrev "RonZhong2009" <notifications@github.com>:

I can't open the your attached file. But from your descriptions, seems we
have encountered the same issue. I upgraded to the latest 3.19.3, the issue
still got hit. Here is my stackframes about it:

Stack: [0x000000006bdf0000,0x000000006bef0000], sp=0x000000006bee8160, free
space=992k
Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native
code)
C [sqlite-3.19.3-1b1aca0b-79af-4746-ac09-72cc0a3c9878-
sqlitejdbc.dll+0x122d4]
C [sqlite-3.19.3-1b1aca0b-79af-4746-ac09-72cc0a3c9878-
sqlitejdbc.dll+0x1f76e]
C 0x00000000045635fb

Java frames: (J=compiled Java code, j=interpreted, Vv=VM code)
J 8999 org.sqlite.core.NativeDB.column_int(JI)I (0 bytes) @
0x00000000045634bb [0x0000000004563480+0x3b]
J 9581 C1 org.sqlite.jdbc3.JDBC3ResultSet.getInt(Ljava/lang/String;)I (10
bytes) @ 0x0000000003a2d4d4 [0x0000000003a2cfa0+0x534]

Any updates regarding this issue? Thanks in advance.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/xerial/sqlite-jdbc/issues/212#issuecomment-311610694>,
or mute the thread

**Comment by RonZhong2009** [comment]
Thanks for your reply.  I think you're right. 
The code which I am working on  did behave like you said, somehow closed the connection before calling accessing rs.getInt(). 
I tried to unify the connections to only one , and ensured that  disconnection is the end.
Then there's not a crash any more.

**Comment by jacob-lerenius** [comment]
Hi,
No problem. This is what I did too. All examples show an open/close on the
connection at each transaction but I also think one unified connection is
the way to avoid tjus problem and also keep up the performance. I use an
external lock when accessing the db, but it may not be nessesary but a good
way to ensure the DB if not closed by a thread in the middle of a
trandaction.
Happy coding :)

Den 29 juni 2017 6:46 em skrev "ron_zhong" <notifications@github.com>:

Thanks for your reply. I think you're right.
The code which I am working on did behave like you said, somehow closed the
connection before calling accessing rs.getInt().
I tried to unify the connections to only one , and ensured that
disconnection is the end.
Then there's not a crash any more.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/xerial/sqlite-jdbc/issues/212#issuecomment-312025032>,
or mute the thread
